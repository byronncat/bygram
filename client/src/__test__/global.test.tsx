import { render } from '@testing-library/react'

import { uri, useGlobalContext, Global, Loader, Overlay } from '@global'
import { useEffect } from 'react'

describe('api', () => {
  describe('server', () => {
    const OLD_ENV = process.env

    beforeEach(() => {
      jest.resetModules() // Most important - it clears the cache
      process.env = { ...OLD_ENV } // Make a copy
    })

    afterAll(() => {
      process.env = OLD_ENV // Restore old environment
    })

    it('should return local uri', () => {
      const { REACT_APP_API_URL, ...rest } = process.env
      process.env = { ...rest }

      const path = uri.getHostingServer('/api/test')
      expect(path).toBe('/api/test')
    })

    it('should return hosting server uri', () => {
      process.env.REACT_APP_API_URL = 'https://example.com'

      const path = uri.getHostingServer('/api/test')
      expect(path).toBe('https://example.com/api/test')
    })
  })

  it('should format image CDN correctly', () => {
    const path = uri.transformImageCDN(
      'https://res.cloudinary.com/demo/image/upload/v1626321412/demo.jpg',
      'w_100,fl_lossy,q_auto'
    )

    expect(path).toBe(
      'https://res.cloudinary.com/demo/image/upload/w_100,fl_lossy,q_auto/demo.jpg'
    )
  })
})

describe('components', () => {
  describe('Loader', () => {
    it('should render the loader correctly', () => {
      const { container } = render(<Loader />)
      expect(container).toMatchSnapshot()
    })
  })

  describe('Overlay', () => {
    it('should render the overlay correctly', () => {
      const { container } = render(<Overlay exitHandler={() => {}} />)
      expect(container).toMatchSnapshot()
    })
  })
})

describe('hocs', () => {
  describe('Global', () => {
    it('should render the global hoc correctly', () => {
      const { container } = render(<Global>Test</Global>)
      expect(container).toMatchSnapshot()
    })

    it('should refresh the page', () => {
      let renderCount = -1
      const Test = () => {
        renderCount++
        const { refreshPage } = useGlobalContext()

        useEffect(() => {
          refreshPage()
        }, [])
        return <div>Test</div>
      }

      render(
        <Global>
          <Test />
        </Global>
      )
      expect(renderCount).toBe(1)
    })
  })
})

describe('pages', () => {
  // it('should render the 404 page correctly', () => {
  //   const { container } = render(
  //     <Router>
  //       <ErrorPage />
  //     </Router>
  //   )
  //   expect(container).toMatchSnapshot()
  // })
})
