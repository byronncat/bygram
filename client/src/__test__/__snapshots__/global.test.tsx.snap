// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Loader should render the loader correctly 1`] = `
<div>
  <div
    class="d-flex flex-column justify-content-center align-items-center w-100 h-100"
  >
    <div
      class="spinner-box"
    >
      <div
        class="configure-border-1"
      >
        <span
          class="configure-core"
        />
      </div>
      <div
        class="configure-border-2"
      >
        <span
          class="configure-core"
        />
      </div>
    </div>
    <div
      class="text waviy mt-5"
    >
      <span>
        L
      </span>
      <span>
        O
      </span>
      <span>
        A
      </span>
      <span>
        D
      </span>
      <span>
        I
      </span>
      <span>
        N
      </span>
      <span>
        G
      </span>
    </div>
  </div>
</div>
`;

exports[`components Overlay should render the overlay correctly 1`] = `
<div>
  <div
    class="position-absolute top-0 start-0 z-1 w-100 h-100 d-flex justify-content-center align-items-center"
  >
    <span
      class="overlay z-n1"
    />
    <button
      aria-label="close"
      class="button shadow-none btn-close btn-close-white position-absolute top-0 end-0 p-3"
      type="button"
    />
  </div>
</div>
`;

exports[`hocs Global should render the global hoc correctly 1`] = `
<div>
  <div
    class="Toastify"
  />
  Test
</div>
`;

exports[`pages should render the 404 page correctly 1`] = `
<div>
  <div
    class="wrapper w-100 h-100 d-flex"
  >
    <div
      class="square404"
      id="Square"
    >
      <span
        class="square flicker-1"
      >
        <p>
          404
        </p>
      </span>
    </div>
    <div
      class="texts d-flex flex-column justify-content-center"
    >
      <h4
        class="fs-4"
      >
        Oops! Page not found.
      </h4>
      <p
        class="pt-3 pb-5"
      >
        The page you are looking for does not exist. Go back to the main page.
      </p>
      <a
        class="button"
        href="/"
      >
        Back to Home
      </a>
    </div>
  </div>
</div>
`;
